addon context (#762)
oam-dev/kubevela.io: docs/platform-engineers/addon/intro.md

        name: my-account
```

##### Use context render component

> Beside use `parameter` to generate component dynamically when enable the addon, you also use context to do this. All field defined in `metadata.yaml` can be used in rendering phase.

Reviewer: ```suggestion
Besides using `parameter` to generate component dynamically, you can also use `context` to  render runtime variable.
```

Reviewer: >  All field defined in `metadata.yaml` can be used in rendering phase.

这句话很不清楚，要搞个表格

For example, you can define the component with cue like this:
```cue
output: {
	type: "webservice"
	properties: {
		image: "oamdev/vela-apiserver:" + context.metadata.version
		....
    }
}
```

And the `metadata.yaml` is:
```yaml
...
name: velaux
version: 1.2.4
...
```

> The render result is:

Reviewer: ```suggestion
The result will be:
```

```yaml
kind: Application
... 
# application header in template
spec:
  components:
  - type: webservice
    properties:
    	image: "oamdev/vela-apiserver:v1.2.4"
```

> The image tag will be set with addon's version. The real example is [VelaUX](https://github.com/kubevela/catalog/blob/master/addons/velaux/resources/apiserver.cue).

Reviewer: ```suggestion
The image tag becomes the addon's version due to the `context.metadata.version` points to. The real example is [VelaUX](https://github.com/kubevela/catalog/blob/master/addons/velaux/resources/apiserver.cue).
```
